### PR TITLE
Generate and save pdf files into database during SendLetter requests

### PR DESCRIPTION
**Pull request by banderous**
### Change description ###

Generate PDF files during the SendLetter REST request and persist them in the database, with a test to validate that the PDF is valid using Apache pdfbox.

I'm sorry for the big diff, in order to keep the tests working I  made a number of other changes including:

* Removing the API endpoints that send letter consumer uses to talk to the producer
* Removed consumer Letter class
* removed consumer SendLetterClient
* removed consumer SendLetterService
* refactored PdfCreator to use producer's Letter class instead of duplicate consumer version

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 1]
```diff
@@ -49,39 +46,29 @@ public UUID send(Letter letter, String serviceName) throws JsonProcessingExcepti
...
-        }
+        return new LetterStatus(id, l.messageId, toDateTime(l.createdAt),
+            toDateTime(l.sentToPrintAt), toDateTime(l.printedAt), l.isFailed);
     }
 
-    public void checkPrintState() {
-        // TODO
-        letterRepository.getStaleLetters();
+    private ZonedDateTime toDateTime(Timestamp stamp) {
+        return stamp.toInstant().atZone(ZoneId.of("UTC"));
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: possible NPE

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreator.java [hunk 2]
```diff
@@ -37,11 +31,10 @@ public PdfCreator(DuplexPreparator duplexPreparator) {
     private byte[] generatePdf(Document document) {
         Instant start = Instant.now();
         byte[] pdf = generatePdf(document.template.getBytes(), document.values);
-        insights.trackPdfGenerator(Duration.between(start, Instant.now()), true);
         return pdf;
     }
 
-    public PdfDoc create(Letter letter) {
+    public byte[] create(Letter letter) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreator.java]: can this be private? if so, maybe just put it in the public method below

Comment (banderous) [reply src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/PdfCreator.java]: The reason for the two methods is that the FTP code from the consumer uses the PdfDoc type that saves the files to disk, whereas the Letter entity uses a byte array.

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java (hunk 1 above)
Comment (banderous) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: good spot

Comment (banderous) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: done

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java [hunk 3]
```diff
@@ -49,39 +46,32 @@ public UUID send(Letter letter, String serviceName) throws JsonProcessingExcepti
...
     }
 
-    public void checkPrintState() {
-        // TODO
-        letterRepository.getStaleLetters();
+    public static ZonedDateTime toDateTime(Timestamp stamp) {
+        if (null == stamp) {
+            return null;
+        }
+        return stamp.toInstant().atZone(ZoneId.of("UTC"));
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java]: you can use `ZoneOffset.UTC` constant

**Pull request merged**